lib/flat-compat.js

```diff
@@ -0,0 +1,298 @@
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            flatConfig.linterOptions = linterOptions;
+            linterOptions[key] = eslintrcConfig[key];
+        }
+    }
+
+    // move ecmaVersion a level up
+    if (languageOptions.parserOptions) {
+
+        if ("ecmaVersion" in languageOptions.parserOptions) {
+            languageOptions.ecmaVersion = languageOptions.parserOptions.ecmaVersion;
+            delete languageOptions.parserOptions.ecmaVersion;
+        }
+
+        if ("sourceType" in languageOptions.parserOptions) {
+            languageOptions.sourceType = languageOptions.parserOptions.sourceType;
+            delete languageOptions.parserOptions.sourceType;
+        }
+
+        // check to see if we even need parserOptions anymore and remove it if not
+        if (Object.keys(languageOptions.parserOptions).length === 0) {
+            delete languageOptions.parserOptions;
+        }
+    }
+
+    if (eslintrcConfig.ignorePattern) {
+        const ignore = IgnorePattern.createIgnore([eslintrcConfig.ignorePattern]);
+
+        configs.unshift({
+
+            /*
+             * We need to wrap the ignore function in another function because the
+             * ignore function's second parameter is a boolean `dot` that was used
+             * in ESLintRC but flat config's second parameter is an options object.
+             * We don't want to pass in the options object when the ignore function
+             * expects a boolean so we just pass in the first argument.
+             */
+            ignores: [absoluteFilePath => ignore(absoluteFilePath)]
+        });
+    }
+
+    // overrides
+    if (eslintrcConfig.criteria) {
+        flatConfig.files = [absoluteFilePath => eslintrcConfig.criteria.test(absoluteFilePath)];
+    }
+
+    // translate plugins
+    if (eslintrcConfig.plugins && typeof eslintrcConfig.plugins === "object") {
+        debug(`Translating plugins: ${eslintrcConfig.plugins}`);
+
+        flatConfig.plugins = {};
+
+        for (const pluginName of Object.keys(eslintrcConfig.plugins)) {
+
+            debug(`Translating plugin: ${pluginName}`);
+            debug(`Resolving plugin '${pluginName} relative to ${resolvePluginsRelativeTo}`);
+
+            const plugin = eslintrcConfig.plugins[pluginName].definition;
+
+            flatConfig.plugins[pluginName] = plugin;
+
+            // create a config for any processors
+            if (plugin.processors) {
+                for (const processorName of Object.keys(plugin.processors)) {
+                    if (processorName.startsWith(".")) {
+                        debug(`Assigning processor: ${pluginName}/${processorName}`);
+
+                        configs.unshift({
+                            files: [`**/*${processorName}`],
+                            processor: plugin.processors[processorName]
+                        });
+                    }
+
+                }
+            }
```

Comment:
The created config is meant to not have any further lookups.

Reply:
Currently, two lookups remain.

- `processor` that end-users gave. (e.g., `processor: "my-plugin/a-processor"`)
- `rules`.

I think we cannot remove the `rules`' lookup.

We can remove the `processor`'s lookup. Currently, the compat layer copies the `processor` setting as-is ([/lib/flat-compat.js#L90-L95](https://github.com/eslint/eslintrc/blob/b8169d695e706995494311778a21cb22f66e0b15/lib/flat-compat.js#L90-L95)). Instead, it can resolve processors from [`configArray.pluginProcessors` property](https://github.com/eslint/eslintrc/blob/b86d0c24b6fc3e941913469355ba77fccb8a47bc/lib/config-array/config-array.js#L437-L444). Therefore, it can address both processors that end-users configured and of "implicit-processor" config array elements.

Reply:
At the moment, that part of the code doesn't have access to the config array, and I'd rather not introduce a new parameter for the sake of a small efficiency gain. What I have here is working correctly, so I'd like to stick with this as a first draft. We can always make changes later, but my main goal right now is to get something working so we can start making changes in the main repo.